### PR TITLE
resource_control: add new config `priority_ctl_strategy` to enhance priority control

### PR DESCRIPTION
**Pull request by glorv**
### What is changed and how it works?

Issue Number: ref #17531

What's Changed:

```commit-message
Add a new config `priority_ctl_strategy` to allow user choose different limit strategy to control low-priority tasks.
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]
lgtm. 

BTW, do u test with SQL interface?

**Review by Connor1996 — APPROVED** [review]
LGTM

**Comment by glorv** [comment]
> lgtm.
> 
> BTW, do u test with SQL interface?

The mechanism with other dynamic configs. And I have manually test it in my local dev environment.

**Comment by glorv** [comment]
@zhangjinpeng87  PTAL

**Comment by glorv** [comment]
@cfzjywxk @zhangjinpeng87  PTAL, thanks

File: components/resource_control/src/config.rs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            enabled: true,
+            priority_ctl_strategy: PriorityCtlStrategy::Moderate,
+        }
+    }
+}
+
+#[derive(Clone, Copy, Default, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum PriorityCtlStrategy {
+    Aggressive,
```
Comment (zhangjinpeng87) [inline components/resource_control/src/config.rs:29]: Can you provide some comments for different option?

**Review by zhangjinpeng87 — APPROVED** [review]
LGTM for adding "priority-ctl-strategy" config for resource control feature.

File: components/resource_control/src/config.rs (hunk 1 above)
Comment (glorv) [reply components/resource_control/src/config.rs:33]: Done.

**Pull request merged**